### PR TITLE
[RELEASE] wallet: fix auto low priority so that it takes effect only when saved…

### PR DESCRIPTION
**Pull request by stoffu**
… default is 0

### CODE REVIEW

File: src/wallet/api/wallet.cpp [hunk 1]
```diff
@@ -1149,7 +1147,7 @@ PendingTransaction *WalletImpl::createTransaction(const string &dst_addr, const
                 de.is_subaddress = info.is_subaddress;
                 dsts.push_back(de);
                 transaction->m_pending_tx = m_wallet->create_transactions_2(dsts, fake_outs_count, 0 /* unlock_time */,
-                                                                          adjusted_priority,
+                                                                          static_cast<uint32_t>(priority),
```
Comment (moneromooo-monero) [inline src/wallet/api/wallet.cpp]: Looks like it's already uint32_t.

File: src/wallet/wallet2.cpp [hunk 2]
```diff
@@ -5106,7 +5107,7 @@ uint64_t wallet2::adjust_mixin(uint64_t mixin) const
 //----------------------------------------------------------------------------------------------------
 uint32_t wallet2::adjust_priority(uint32_t priority)
 {
-  if (priority == 0 && get_default_priority() != 1 && auto_low_priority())
+  if (priority == 0 && m_default_priority == 0 && auto_low_priority())
```
Comment (moneromooo-monero) [inline src/wallet/wallet2.cpp]: I don't get why this line is not the only change needed. Can you explain ?

Comment (stoffu) [reply src/wallet/wallet2.cpp]: You're right, I was somehow missing that and overcomplicating things. Thanks!

File: src/wallet/api/wallet.cpp (hunk 1 above)
Comment (stoffu) [reply src/wallet/api/wallet.cpp]: It's `PendingTransaction::Priority` fwiw.

**Pull request merged**